activesupport/lib/active_support/claims.rb

```diff
@@ -0,0 +1,46 @@
+require 'active_support/time'
+
+module ActiveSupport  
+  class Claims
+    class InvalidClaims < StandardError; end
+    class ExpiredClaims < StandardError; end
+    
+    attr_reader :payload, :purpose, :expires
+
+    def initialize(options)
```

Comment:
Why not do it like this?

```ruby
def initialize(value:, **options)
  @payload = value
  @purpose = self.class.pick_purpose(options)
  @expires_at = self.class.pick_expiration(options)
end
```

Then we can still defer the default expiry decision to some time later.